apps/server/src/modules/board/uc/element.uc.ts

```diff
@@ -30,6 +32,36 @@ export class ElementUc {
 		await this.elementService.update(element, content);
 	}
 
+	async createSubmissionItem(userId: EntityId, contentElementId: EntityId): Promise<SubmissionItem> {
+		const submissionContainer = (await this.elementService.findById(contentElementId)) as SubmissionContainerElement;
+
+		if (!(submissionContainer instanceof SubmissionContainerElement))
+			throw new HttpException(
+				'Cannot create submission-item for non submission-container-element',
+				HttpStatus.UNPROCESSABLE_ENTITY
+			);
+
+		if (!submissionContainer.children.every((child) => child instanceof SubmissionItem))
+			throw new HttpException(
+				'Children of submission-container-element must be of type submission-item',
+				HttpStatus.UNPROCESSABLE_ENTITY
+			);
+
+		const userIds = submissionContainer.children.flatMap((item) => (item as SubmissionItem).userId);
```

Comment:
maybe a filter and checking the length is easier to read. Not sure regarding performance...

Reply:
performance will be no issue, I simplified to query with a find